python/lib/client/dmod/client/request_clients.py

```diff
@@ -175,6 +202,50 @@ async def delete_dataset(self, name: str, **kwargs) -> bool:
         self.last_response = await self.async_make_request(request)
         return self.last_response is not None and self.last_response.success
 
+    async def get_dataset_content_details(self, name: str, **kwargs) -> dict:
+        # TODO: later add things like created and last updated perhaps
+        query = DatasetQuery(query_type=QueryType.GET_DATASET_ITEMS)
+        request = DatasetManagementMessage(action=ManagementAction.QUERY, query=query, dataset_name=name)
+        self.last_response: DatasetManagementResponse = await self.async_make_request(request)
+        if self.last_response.success:
+            return self.last_response.data
+        else:
+            return {}
```

Comment:
should we return something to the caller indicating why, if we know, the response was unsuccessful?